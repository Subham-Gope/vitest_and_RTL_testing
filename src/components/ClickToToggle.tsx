import React, { useState } from "react";
import type { MouseEvent } from "react";

interface PropType {
  message: string;
}

const ClickToToggle: React.FC<PropType> = ({ message }) => {
  const [show, setShow] = useState<boolean>(false);

  const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
    setShow((prev) => !prev);
  };

  return (
    <>
      <h1>{message}</h1>
      <button onClick={handleClick}>Click Me</button>
      {show && <p>Hello there, now the text is visible!</p>}
    </>
  );
};

export default ClickToToggle;
