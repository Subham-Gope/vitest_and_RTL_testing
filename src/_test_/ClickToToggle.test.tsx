import { describe, test, expect } from "vitest";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ClickToToggle from "../components/ClickToToggle";

describe("Testing the ClickToToggle.tsx file", () => {
  test("clicks a button to toggle the text visibility", async () => {
    // Setting up the userEvent
    const user = userEvent.setup();

    const { container } = render(<ClickToToggle message={"Hello Guys"} />);
    expect(container).toMatchSnapshot();

    const button = screen.getByRole("button", { name: /click me/i });
    expect(button).toBeInTheDocument();

    // Initiating user interaction
    await user.click(button);

    const visibleText = screen.getByText(/text is visible/i);
    expect(visibleText).toBeInTheDocument();
  });
});
