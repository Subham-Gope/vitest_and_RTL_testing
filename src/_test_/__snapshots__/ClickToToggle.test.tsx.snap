// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ClickToToggle.tsx file > clicks a button to toggle the text visibility 1`] = `
<div>
  <h1>
    Hello Guys
  </h1>
  <button>
    Click Me
  </button>
</div>
`;
